src/main.rs

```diff
@@ -67,20 +67,24 @@ fn parse_input() {
     }
 }
 
-fn format_pair(pair: Pair<&str>, indent_level: u32) -> String {
-    let mut indent = String::new();
-
-    for _ in 0..indent_level {
-        indent.push_str("  ");
-    }
+fn format_pair(pair: Pair<&str>, indent_level: u32, is_newline: bool) -> String {
+    let indent = if is_newline {
+        let mut indent = String::new();
```

Comment:
You should use [`str::repeat`](https://doc.rust-lang.org/std/primitive.str.html#method.repeat) here instead.